examples/with-segment-analytics/components/analytics.tsx

```diff
@@ -0,0 +1,23 @@
+'use client'
+
+import { useEffect } from 'react'
+import { usePathname, useSearchParams } from 'next/navigation'
+import { AnalyticsBrowser } from '@segment/analytics-next'
+
+// This write key is associated with https://segment.com/nextjs-example/sources/nextjs.
+const DEFAULT_WRITE_KEY = 'NPsk1GimHq09s7egCUlv7D0tqtUAU5wa'
+
+export const analytics = AnalyticsBrowser.load({
+  writeKey: process.env.NEXT_PUBLIC_SEGMENT_WRITE_KEY || DEFAULT_WRITE_KEY,
+})
+
+export default function Analytics() {
```

Comment:
Wouldn't it be more idiomatic for this to be a hook like `useAnalyticsPageEvent()`?

Reply:
I added the hook to the component, should I add it to a separate file? E.g. `lib/useSegment.ts`

Reply:
Ah, I see:
```
Error: Attempted to call useAnalyticsPageEvent() from the server but useAnalyticsPageEvent is on the client. It's not possible to invoke a client function from the server, it can only be rendered as a Component or passed to props of a Client Component.
```

You cannot make the Root Layout a Client Component. So you will still need the `<Analytics />` component. But you can call the hook inside of it.

Reply:
<img width="1460" alt="Screenshot 2023-08-30 at 12 12 24 PM" src="https://github.com/vercel/next.js/assets/320341/f588434d-86f9-473b-b943-391e18357d8e">

Great Example, btw @lukebussey! Followed it and got it working in my Next13 App Router App :)